src/main/kotlin/de/bund/digitalservice/useid/config/SecurityHeadersFilter.kt

```diff
@@ -31,10 +32,13 @@ class ContentSecurityPolicyFilter(
         val hostNameIsAllowed = hostName?.let { contentSecurityPolicyProperties.domainIsAllowed(it) }
 
         if (hostNameIsAllowed == true) {
-            response.headers.set(
-                "Content-Security-Policy",
-                contentSecurityPolicyProperties.getCSPHeaderValue(hostName)
+            val securityHeaders = mapOf(
+                "Content-Security-Policy" to contentSecurityPolicyProperties.getCSPHeaderValue(hostName),
+                HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN to hostName,
+                HttpHeaders.VARY to HttpHeaders.ORIGIN
```

Comment:
Please refer to MDN:
https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#the_http_response_headers 

> If the server specifies a single origin (that may dynamically change based on the requesting origin as part of an allowlist) rather than the "*" wildcard, then the server should also include Origin in the [Vary](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Vary) response header to indicate to clients that server responses will differ based on the value of the [Origin](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin) request header.